integrations/os2sync/os2mo.py

```diff
@@ -198,6 +200,29 @@ def addresses_to_orgunit(orgunit, addresses):
             orgunit["Post"] = a["value"]
 
 
+def kle_to_orgunit(orgunit, kle):
+    """
+    Aspect Udførende goes into Tasks
+    Aspect Ansvarlig goes into ContactForTasks
+    """
+    tasks = []
+    contactfortasks = []
+
+    for k in kle:
+        uuid = k["uuid"]
+        for a in k["kle_aspect"]:
+            if a["scope"] == "UDFOERENDE":
+                tasks.append(uuid)
+            elif a["scope"] == "ANSVARLIG":
+                contactfortasks.append(uuid)
+
+    if len(tasks):
+        orgunit["Tasks"] = list(set(tasks))
+
+    if len(contactfortasks):
+        orgunit["ContactForTasks"] = list(set(contactfortasks))
```

Comment:
Denne metode havde jeg nok grebet lidt anderledes an:
```suggestion
    # Construct a set of tuples (uuid, kle aspect)
    # {('uuid1', 'UDFOERENDE'), ('uuid3', 'UDFOERENDE'), ('uuid2', 'ANSVARLIG'), ('uuid3', 'ANSVARLIG')}
    tuple_set = {(kle['uuid'], kle_aspect['scope']) for kle in kles for kle_aspect in kle['kle_aspect']}
    # Filter + pickout uuid
    orgunit["Tasks"] = list({uuid for uuid, aspect in setted if aspect == 'UDFOERENDE'})
    orgunit["ContactForTasks"] = list({uuid for uuid, aspect in setted if aspect == 'ANSVARLIG'})
```

Reply:
Alternativt til at lave to gennemløb for filtering, kan et enkelt laves vha. `groupby` fra `itertools`:
```
    from itertools import groupby
    import operator

    # Partition the set according to kle_aspect (tuple[1])
    groups = groupby(tuple_set, key=operator.itemgetter(1))
    
    # Convert groups to dicts of sets
    group_dict = {aspect: {uuid for (uuid, aspect) in item_iterator} for (aspect, item_iterator) in groups}
    
    orgunit["Tasks"] = list(group_dict.get("UDFOERENDE", []))
    orgunit["ContactForTasks"] =  list(group_dict.get("ANSVARLIG", []))
```
Men det bliver nok lidt langhåret, og svært at forstå.

Reply:
Som minimum ville jeg nok lave metoden, således at:
```
tasks = []
contactfortasks = []
```
I stedet var sets direkte, således at man ikke bygger en stor liste for at lave den til et set, for at lave den til en liste igen. Så kan vi ligeså godt bygge et set direkte.

Reply:
Ja, jeg går efter at skrive min kode så lavpandet, at alle burde kunne forstå den. Det har to kæmpe fordele:

1) Det er nemt for andre at finde mine fejl.
2) Jeg kan også selv forstå den.

Og jeg kan ikke få øje på bagdelene. Jeg ved godt at der findes smartness, men jeg har bestemt ikke behov for at bruge det, med mindre der er en vanvittigt god grund (somme tider kan smartness faktisk gøre kompleks kode mere læsbar)

Reply:
> Som minimum ville jeg nok lave metoden, således at:
> 
> ```
> tasks = []
> contactfortasks = []
> ```
> 
> I stedet var sets direkte, således at man ikke bygger en stor liste for at lave den til et set, for at lave den til en liste igen. Så kan vi ligeså godt bygge et set direkte.

Jeg har aldrig været vant til at bruge sets som 'lister', jeg vidste faktisk ikke man kunne. Det retter jeg lige op